### ISSUE TITLE
Crash debugging Vulkan applications built with SDK 1.0.8

### ISSUE THREAD

**Issue by Honeybunch** [baldurk/renderdoc#247]
I recently installed and updated a project of mine to use the 1.0.8 LunarG SDK for Vulkan and I can no longer properly debug with Renderdoc. I can debug the same application on 1.0.5 but not the latest. 
I'm using a GTX 780 with Driver version 364.91 on Windows 10. Renderdoc is the absolute latest; I built it from source just a few minutes before writing this. 

The crash happens with my application, [Hatchit](https://github.com/thirddegree/Hatchit/tree/dev) as well as any of the examples from [Sascha Willems](https://github.com/SaschaWillems/Vulkan) but ONLY when Validation is enabled. If you take any example from Sascha Willems and turn on validation it will break. If you leave validation off this repro will not cause a crash. 

The repro is simple:
* Attempt to capture any executable using the 1.0.8 SDK that has validation enabled
* Capture a frame
* Load the log
* Go to the texture viewer
* RenderDoc immediately crashes

Running RenderDoc in debug mode the crash happens at line 1212 of vk_replay.cpp with the error:
```
Exception thrown at 0x00007FF9F7C42F6C (VkLayer_core_validation.dll) in renderdocui.exe: 0xC0000005: Access violation reading location 0x0000000000000000.
```
Here is a picture of the callstack:
![CallStack](http://i.imgur.com/SVTWRSn.png)

The new errors I get in my app and with Sascha Willem's examples with the 1.0.8 SDK are:
```
ERROR: [SC] Code 15 : Shader requires VkPhysicalDeviceFeatures::shaderClipDistance but is not enabled on the device
ERROR: [SC] Code 15 : Shader requires VkPhysicalDeviceFeatures::shaderCullDistance but is not enabled on the device
```
Which I haven't found any solution for an which only show up on the Nvidia 364.91 driver. (I don't think AMD has a 1.0.8 compliant driver anyway). The cullDistance error goes away when using a shader version below `#version 450` but the clipDistance error shows up on every shader I've written. 

The validation layer names changed with 1.0.8. Could this have something to do with it?

**Comment by baldurk** [comment]
The crash lies in the validation layers, not in renderdoc - I think this case is one I've seen before, it's because of something I'm doing that's not spec valid and the validation layers trip up and hit a NULL pointer.

I've been meaning to fix it for a while, which should hopefully sort the issue, but in the meantime disabling validation while capturing renderdoc should work as a workaround.

As to the cullDistance/clipDistance errors - I'm not quite clear on what you're getting at. Are you saying that you only get these errors when running with renderdoc?

**Comment by Honeybunch** [comment]
I get the clip/cull errors with everything but they were new with 1.0.8 so I wasn't sure if they were relevant.

**Comment by baldurk** [comment]
Ah right. No I think they were just a new check added in 1.0.8. I found that just removing those declarations from gl_PerVertex did the trick - b630b84e2026e8169bc0d84ad0ae067f969c8859

**Comment by Honeybunch** [comment]
Oh yep that did it. Thanks. But yeah it looks like those errors were unrelated to the crash anyway.

**Comment by DominikWitczakAMD** [comment]
Honeybunch: The ".8" part defines a patch number, which technically means that any changes made at this level are backward-compatible and should not affect applications. All in all, this means there's no such thing as 1.0.8-compliancy (for the lack of better wording ..). This has been made clear in the latest spec version (https://www.khronos.org/registry/vulkan/specs/1.0/xhtml/vkspec.html )

Just thought I'd chime in and clarify.

**Comment by DominikWitczakAMD** [comment]
Oh, by the way - you may want to consider building the layers yourself (https://github.com/KhronosGroup/Vulkan-LoaderAndValidationLayers ) and checking where the crash is coming from. If this turns out to be a problem with the validation layer, it would be much appreciated if you could report the error to the team by creating an issue there.

**Comment by baldurk** [comment]
If it's the crash I'm thinking of, it's because the texture display shader declares 12 different samplers and samples from only one via dynamic branching. I'm only updating the one relevant descriptor - the others are uninitialised. The validation layers definitely crashed out on that at some point, but I think on latest master branch they're working and giving an explicit error for this case.

As I mentioned above, I've been dragging my heels on fixing it since it means creating a dozen different dummy minimum-sized textures and updating them constantly into the descriptors that will never be read. It's been lower priority since it's not actually a problem on any of the platforms I've tested. It's still spec invalid though so I need to fix it.

**Comment by Honeybunch** [comment]
@DominikWitczakAMD Thanks for the clarification.

**Comment by baldurk** [comment]
I believe that commit should fix the naughty behaviour, and the 1.0.8 validation layers shouldn't crash anymore.

**Comment by Honeybunch** [comment]
Just tested it and it works fine. Thanks so much!
